ddtrace/contrib/pytest/plugin.py

```diff
@@ -306,111 +310,132 @@ def _get_test_class_hierarchy(item):
     return ".".join(test_class_hierarchy)
 
 
-@pytest.hookimpl(tryfirst=True, hookwrapper=True)
-def pytest_runtest_protocol(item, nextitem):
-    if not _CIVisibility.enabled:
-        yield
-        return
-    test_session_span = _extract_span(item.session)
+def get_path(item):
+    if hasattr(item, "path"):
+        return str(item.path)
+    parent = item
+    while not hasattr(parent, "fspath"):
```

Comment:
I was thinking that if we get to a point where `parent.parent == None`, we'll end up with ```AttributeError: 'NoneType' object has no attribute 'parent'```, but I actually think that's okay because it would be a weird enough scenario that raising an exception makes sense.

Reply:
It can happen, maybe for runtime generated tests. I agree that this could raise the exception but I'd rather capture it and log it from the calling function

Reply:
I went with a solution to not reinvent the wheel